tiledb/sm/filesystem/posix.cc

```diff
@@ -455,23 +437,18 @@ Status Posix::read(
         std::string("Cannot read from file ' ") + path.c_str() +
         "'; offset > typemax(off_t)"));
   }
-  if (nbytes > SSIZE_MAX) {
+  if (nbytes > SIZE_MAX) {
     return LOG_STATUS(Status_IOError(
-        std::string("Cannot read from file ' ") + path.c_str() +
-        "'; nbytes > SSIZE_MAX"));
-  }
-  uint64_t bytes_read = read_all(fd, buffer, nbytes, offset);
-  if (bytes_read != nbytes) {
-    return LOG_STATUS(Status_IOError(
-        std::string("Cannot read from file '") + path.c_str() +
-        "'; File reading error"));
+        std::string("Cannot read from file ' ") + path +
+        "'; nbytes > SIZE_MAX"));
   }
+  Status st = read_all(fd, buffer, nbytes, offset);
   // Close file
   if (close(fd)) {
     return LOG_STATUS(Status_IOError(
         std::string("Cannot read from file; ") + strerror(errno)));
   }
```

Comment:
Since routine now returning a status, 
if both read_all() and close() failed,
is this perhaps returning a lesser error than might have been returned from the read_all()?

Reply:
I thought of just logging the error and not failing if we can't close the handle; we will have to do so when we move the code to exceptions.

Reply:
We should log the error from close but otherwise ignore it. Errors from close while rare are important, but mostly on the write side where we might not have had data reach disk.

Reply:
Occasionally some code paths (can't remember if I've seen in tiledb) will actually check previous status, when there are potentially two conflicting ones, and if the 'primary' status is bad only log the secondary failing status (close in this case) and return the primary (Failing) status, while if the primary status was good will instead log and return the secondary status.